tsconfig.json

```diff
@@ -1,11 +1,19 @@
 {
   "compilerOptions": {
+    "incremental": true,
     "strict": true,
-    "noEmit": true
+    "noImplicitAny": true,
+    "declaration": true,
+    "outDir": "typings",
+    "target": "es2017"
```

Comment:
Do I understand correctly this is a breaking change, since you're dropping support for engines that don't have es2017 support? (looking at you, ie11)

Reply:
I can set this to `es5` to be less dramatic (and also support IE11), but in fact we already dropped integration tests for IE11 (switched to Edge instead). According to https://caniuse.com/usage-table then IE11 and lesser versions account for 0.73% of all browsers which is not much but not exactly a little either.

One benefit of dropping it is smaller bundle sizes, es5 bundle sizes are a bit large. But maybe that day should come later rather than sooner?

Reply:
I'm totally in to drop ie11, no need to sell me :p~ but it _is_ a major bump

Reply:
decided to not drop es5 in this pr, this change is already big.